app/Http/Controllers/Admin/RecommendedPlanController.php

```diff
@@ -0,0 +1,276 @@
+        if($request->input('name_en') != '---')
+            $all_places->where('name_en', $request->input('name_en'));
+
+        if($request->input('area') != '---')
+            $all_places->where('area_id', $request->input('area'));
+
+        if($request->input('prefecture') != '---')
+            $all_places->where('prefecture_id', $request->input('prefecture'));
+        
+        if($request->input('city') != '---')
+            $all_places->where('city_id', $request->input('city'));
+        
+        return view('admin.plans.place.place_create' ,[
+                'all_places_options' => $all_places_options,
+                'all_places'=> $all_places->paginate(10),
+                'all_areas' => $all_areas,
+                'all_prefectures' => $all_prefectures,
+                'all_cities' => $all_cities,
+                's_area'=> $s_area ?? [],
+                's_prefectures'=> $s_prefecture ?? [],
+                's_city'=> $s_city ?? [] 
+        ]);            
+    }
+
+    //Create the content of the plan
+    public function storeDetail(Request $request, PlanDetail $plan_detail){
+
+        $request->validate([
+            'place_id'  => 'required',
+            'day' => 'required',
+            'sort_no' => 'required',
+            'plan_id' => 'required'
+        ]);
+        
+        $plan_detail = new PlanDetail();
+        $plan_detail->plan_id = $request->plan_id;
+        $plan_detail->place_id = $request->place_id;
+        $plan_detail->day = $request->day;
+        $plan_detail->sort_no = $request->sort_no;
+        $plan_detail->save();
+        
+        return redirect()->route('show.plan_details', ['plan' => $request->plan_id]);
+    }
+        
+    // Use this to display the edit page       
+    public function editDetail(PlanDetail $plan_details, $place_id){
+        $all_places = Place::all();
+        $plans = Plan::all();
+
+        return view('admin.plans.plan_details.recommended_plan_details_edit', [
+            'plans' => $plans,
+            'all_places' => $all_places,
+            'plan_details' => $plan_details,
+            'place_id' => $place_id
+        ]);
+    }
+
+    # -----For display Place list page (for edit page)-----
+    public function updatePlace($plan_detail_id){
+        $all_places_options = Place::all();
+        $all_places = Place::latest()->paginate(10);
+        $all_areas = Area::all();
+        $all_prefectures = Prefecture::all();
+        $all_cities = City::all();
+        $plans = Plan::all();
+
+    return view('admin.plans.place.place_edit')
+        ->with('all_places_options', $all_places_options)
+        ->with('all_places', $all_places)
+        ->with('all_areas', $all_areas)
+        ->with('all_prefectures', $all_prefectures)
+        ->with('all_cities', $all_cities)
+        ->with('plan_detail_id', $plan_detail_id);
+}
+
+    // use this to update the contents in plan
+    public function updateDetail(Request $request, PlanDetail $plan_details){
```

Comment:
my comment above about the `$plan_details` must be singular still applies on this function